Sources/TuistCache/GraphMappers/CacheGraphMutator.swift

```diff
@@ -32,61 +33,96 @@ class CacheGraphMapper: CacheGraphMapping {
         var visitedXCFrameworkPaths: [TargetNode: VisitedXCFramework?] = [:]
         var loadedXCFrameworks: [AbsolutePath: XCFrameworkNode] = [:]
         var sourceTargets: Set<TargetNode> = Set()
 
-        func mapDependencies(_ dependencies: [GraphNode], sourceTargets: inout Set<TargetNode>) throws -> [GraphNode] {
-            var newDependencies: [GraphNode] = []
-            try dependencies.forEach { dependency in
-                // If the dependency is not a target node we keep it.
-                guard let targetDependency = dependency as? TargetNode else {
-                    newDependencies.append(dependency)
-                    return
-                }
-                // If the target cannot be replaced with its associated .xcframework we return
-                guard let xcframeworkPath = xcframeworkPath(target: targetDependency,
-                                                            xcframeworks: xcframeworks,
-                                                            visitedXCFrameworkPaths: &visitedXCFrameworkPaths)
-                else {
-                    sourceTargets.formUnion([targetDependency])
-                    newDependencies.append(dependency)
-                    return
-                }
-
-                // We load the xcframework
-                let xcframework = try self.loadXCFramework(path: xcframeworkPath, loadedXCFrameworks: &loadedXCFrameworks)
-                try mapDependencies(targetDependency.dependencies, sourceTargets: &sourceTargets).forEach { dependency in
-                    if let frameworkDependency = dependency as? FrameworkNode {
-                        xcframework.add(dependency: XCFrameworkNode.Dependency.framework(frameworkDependency))
-                    } else if let xcframeworkDependency = dependency as? XCFrameworkNode {
-                        xcframework.add(dependency: XCFrameworkNode.Dependency.xcframework(xcframeworkDependency))
-                    } else {
-                        // Static dependencies fall into this case.
-                        // Those are now part of the precompiled xcframework and therefore we don't have to link against them.
-                    }
-                }
-                newDependencies.append(xcframework)
-            }
-            return newDependencies
-        }
-
-        func visit(targetNode: TargetNode, sourceTargets: inout Set<TargetNode>) throws {
-            sourceTargets.formUnion([targetNode])
-            targetNode.dependencies = try mapDependencies(targetNode.dependencies, sourceTargets: &sourceTargets)
-        }
         let entryNodes = Set(graph.entryNodes)
         try graph.targets.flatMap { $0.value }
             .filter { targetNode in
                 // We traverse the graph from nodes that are either entry nodes, or are targets that depend on XCTest.
                 entryNodes.contains(targetNode) || (targetNode.dependsOnXCTest && !targetNode.target.product.testsBundle)
```

Comment:
I know it's not a change from this PR but I don't understand the logic here maybe you can help me: why would you exclude the non XCTest targets and targets not being entry nodes?

Reply:
Mmm, I don't think we need it at all.